src/api/app/models/package.rb

```diff
@@ -36,6 +36,9 @@ class UnknownObjectError < APIException
   class ReadSourceAccessError < APIException
     setup 'source_access_no_permission', 403, 'Source Access not allowed'
   end
+  class IllegalFileName < APIException; setup 'invalid_file_name_error'; end
+  class PutFileNoPermission < APIException; setup 403; end
```

Comment:
I think we should move those API exceptions to the controller some day. The sooner the better